membership_prorrate/models/account_invoice.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api, fields
+from datetime import timedelta
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    def _get_membership_interval(self, product, date):
+        """Get the interval to evaluate as the theoretical membership period.
+        :param product: Product that defines the membership
+        :param date: date object for the requested date (if needed for
+        inherited computations)
+        :return: A tuple with 2 date objects with the beginning and the
+        end of the period
+        """
+        date_from = fields.Date.from_string(product.membership_date_from)
+        date_to = fields.Date.from_string(product.membership_date_to)
+        return date_from, date_to
+
+    def _prepare_invoice_line_prorrate_vals(self, invoice_line):
+        product = invoice_line.product_id
+        date_invoice = fields.Date.from_string(
+            invoice_line.invoice_id.date_invoice or fields.Date.today())
+        date_from, date_to = self._get_membership_interval(
+            product, date_invoice)
+        if date_invoice < date_from:
```

Comment:
That's because there's one auto-installed module to join both that I will publish next, that overwrites `_get_membership_interval` to handle the situation with prorrate, so no problem in this one.